src/github/pullRequestGitHelper.ts

```diff
@@ -188,14 +189,20 @@ export class PullRequestGitHelper {
 			const branchName = branchInfos[0].branch!;
 			progress.report({ message: vscode.l10n.t('Checking out branch {0}', branchName) });
 			await repository.checkout(branchName);
-			const remote = readConfig(`branch.${branchName}.remote`);
-			const ref = readConfig(`branch.${branchName}.merge`);
-			progress.report({ message: vscode.l10n.t('Fetching branch {0}', branchName) });
-			await repository.fetch(remote, ref);
+
+			// respect the git setting to fetch before checkout
+			if (vscode.workspace.getConfiguration(GIT).get<boolean>(PULL_BEFORE_CHECKOUT, false)) {
```

Comment:
I may have misunderstood discussion in [thread](https://github.com/microsoft/vscode-pull-request-github/issues/4202). Nevertheless, I have implemented changes you have described, added new setting and it's description

Reply:
@unknovvn the discussion in https://github.com/microsoft/vscode-pull-request-github/issues/4202 covers a lot of ground. All that aside, I think we shouldn't change the default, which is to fetch before checking out a PR branch. Since the default of `git.pullBeforeCheckout` is `false` we need to have a new setting. Thanks for making the changes!